Initiate setup notebook for introduction module

sjspielman opened the pull request:
Closes #563 

This PR adds a starting point notebook for the introduction module. It's a little draft-y, but I'm going to not set it as a draft since we have lots to do!

I explored the use of two 10X datasets with RNA and ADT counts - 
- [NOT USED] The `MALT` data from this link: https://support.10xgenomics.com/single-cell-gene-expression/datasets/3.0.0/malt_10k_protein_v3
  - This data turned out to have _wildly low_ ADT counts that were often comparable to the control Igs, so this is not used in the notebook:
```
             mean detected
CD3    1.88323944 2.574586
CD4    1.46554877 2.310136
CD8a   0.49656859 1.537437
CD14   0.10751610 1.306822
CD15   0.14529999 1.347470
CD16   0.12913973 1.298757
CD56   0.21097841 1.411548
CD19   1.01046347 1.805874
CD25   0.21357743 1.350075
CD45RA 0.52020875 1.636894
CD45RO 0.86705932 1.902918
PD-1   0.99602642 1.936958
TIGIT  0.21054941 1.332533
CD127  0.21241067 1.352356
IgG2a  0.10668592 1.265100 
IgG1   0.14653209 1.306307 
IgG2b  0.03636164 1.096399 
```

- [CURRENTLY USED] PBMC data from this link: https://www.10xgenomics.com/resources/datasets/10-k-human-pbm-cs-with-total-seq-b-human-tbnk-antibody-cocktail-3-v-3-1-3-1-standard-6-0-0

The notebook takes the following overall steps:
- ...But first what it does not do - it does _not_ `wget` or `curl` for you, but there's a markdown chunk about how you can do it (maybe I should have made this a shell chunk). I figure I will make a README and/or script for downloading this once we definitely agree on the final dataset to use.
- per-cell and per-feature QCs and associated filtering for empty cells
- SCE wrangling to split out the ADT counts into an altExp
- Filters RNA and ADT separately:
  - RNA is filtered with `emptyDrops` _or_ `miQC`. That said, `miQC` was extremely hit or miss for me (either convergence or infinite loop 😬 ), so either I'll need some feedback on running `miQC` or we might want to identify a different dataset that plays better with `miQC`? 
    - Related, there is a currently an empty chunk where filtered results between approaches could be compared, pending getting `miQC` up and running. 
  - ADT filtering was an interesting journey! There is a control antibody, so first I tried to use `DropletUtils::cleanTagCounts()` and specify the controls. (See [OSCA](http://bioconductor.org/books/3.16/OSCA.advanced/integrating-with-protein-abundance.html#applying-custom-qc-filters)). However this resulted in bonkers `NA`s in the `discard` column, so I filtered based on the `zero.ambient` column.
- Normalizes RNA and ADT separately:
  - RNA is normalized with the usual `scran` functions
  - ADT journey continues! Following OSCA for cases with control antibodies, I used `scater::librarySizeFactors()` and specified the controls....and the vast majority of cells ended up with size factors of 0. Unsure of how to deal with this, I just, uh, moved on to the `DropletUtils::ambientProfileBimodal()` and `scater::medianSizeFactors()` approach, which led to all non-0 size factors. However, I did not remove the control antibody during ambient calculations, and I wonder if I should have! Lots to discuss here.
- PCA with HVGs and a UMAP, with a quick k-means to give the people their colors
- Cursory cell type annotation with `SingleR` using only one reference. There's a heatmap and UMAP colored by predicted celltypes.   

Finally I have a general question about developing training materials - I ran this notebook without any special versioning considerations, but perhaps I should be doing so inside a particular docker container?

jashapiro commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
```diff
@@ -0,0 +1,272 @@
+  subsets = list(mito = mito_genes))
+
+# we'll need this later, so pop it in now
+pbmc_sce_percellqc <- scater::addPerFeatureQC(pbmc_sce_percellqc)
```
This should happen after filtering, as values change with filtering.

sjspielman replied:
Ah, I realize I meant to delete this since it is not strictly needed for ADT normalization for this data set (it was when I was exploring the MALT data).

jashapiro commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
````diff
@@ -0,0 +1,272 @@
+Before proceeding with further filtering, we'll want to re-arrange the SCE object to handle the RNA and ADT separately.
+Currently, the ADTs and genes are both part of the primary experiment, so let's move the ADT counts into an altExp slot.
+
+```{r}
+gene_names <- as.data.frame(rowData(pbmc_sce_percellqc)) %>% 
+  dplyr::filter(Type == "Gene Expression") %>% 
+  dplyr::pull(ID)
+adt_names <- as.data.frame(rowData(pbmc_sce_percellqc)) %>% 
+  dplyr::filter(Type == "Antibody Capture") %>% 
+  dplyr::pull(ID)
+
+adt_pbmc <- pbmc_sce_percellqc[rownames(pbmc_sce_percellqc) %in% adt_names]
+
+# split out the experiments:
+pbmc_sce_percellqc <- pbmc_sce_percellqc[rownames(pbmc_sce_percellqc) %in% gene_names]
+altExp(pbmc_sce_percellqc, "CITEseq") <- adt_pbmc
+```
````
This should be doable much more simply with [splitAltExps()](https://rdrr.io/bioc/SingleCellExperiment/man/splitAltExps.html), which I think is what we will want to show.

sjspielman replied:
TIL about `splitAltExps()` :tada:!

jashapiro commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
````diff
@@ -0,0 +1,272 @@
+Mito QC stats are already present in `pbmc_sce_percellqc`.
+This is sometimes working and sometimes not for me, so commenting out for now...
+```{r}
+#miqc_model <- miQC::mixtureModel(pbmc_sce_percellqc) 
````
We will want to do both; they serve different functions. First filter empty droplets with emptydops, then compromised (dead) cells with miQC. So miQC should use the filtered data as input.

sjspielman replied:
This fixes itself once the right input is provided!

jashapiro commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
```diff
@@ -0,0 +1,272 @@
+# Filter ADT
+# http://bioconductor.org/books/3.16/OSCA.advanced/integrating-with-protein-abundance.html#applying-custom-qc-filters
+
+controls <- grep("^Ig", rownames(altExp(filtered_sce)))
```
Do these have a specific name? Or are there more controls than just the one? (I see that this comes right from the book, but I don't like using grep if it isn't needed)

sjspielman replied:
There is only 1, and it started with `"IG"`, so yes I ended up copying from the book. I can see definitely how it's easy to look at this and just assume controls are always labeled Ig, so I'll change this to use an explicit vector of the control ADT names.

jashapiro commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
```diff
@@ -0,0 +1,272 @@
+controls <- grep("^Ig", rownames(altExp(filtered_sce)))
+adt_qc_df <- DropletUtils::cleanTagCounts(altExp(filtered_sce), controls=controls) # I feel like this control might be crap!
+
+summary(adt_qc_df$high.controls) # these are all NA????????
+summary(adt_qc_df$discard) # these are MOSTLY NA????????
```
Definitely seems worth investigating this... I'll come back to it when I have better computational resources.

jashapiro commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
```diff
@@ -0,0 +1,272 @@
+filtered_sce <- scran::computeSumFactors(filtered_sce, clusters = qclust)
+norm_sce <- scater::logNormCounts(filtered_sce)
+
+# Normalize ADT
+
+# Calculate baseline:
+# Try out control-based normalization here 
+#  http://bioconductor.org/books/3.16/OSCA.advanced/integrating-with-protein-abundance.html#control-based-normalization
+control_sf <- scater::librarySizeFactors(altExp(norm_sce), subset_row=controls)  # `controls` was previously defined during filtering
+sum(control_sf == 0) # _HOLY WOW_
```
We may not want to get into this in this notebook; I want to show the basics of ADT reading here, but not necessarily spend much time on the interpretation. We aren't going to do integration between RNA & ADT, I don't expect.

jashapiro commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
```diff
@@ -0,0 +1,272 @@
+
+## Cell type annotation
+
+We can introduce annotation with `SingleR`, and it might be a good idea to show how the results are highly influenced by the reference. 
```
Before SingleR, we might want an example of annotation with ADT tags. At least we should plot an ADT tag onto the UMAP.

sjspielman replied:
This is something I haven't been sure about, for a couple reasons..
- Trying to find a good reproducible resource (as in, not via googling each protein name) for what celltypes the tags are roughly markers for
- Also a method has been elusive... or is the idea just to say "highest expressed tag wins" and call it a day?

allyhawkins replied:
I would start with what Josh suggested and do something like plot CD3 and CD19 and say "look these should be B cells" and then with CD4 and CD8 "these should be T cells". I feel like that should be plenty here. 

From the [product sheet of the antibody panel](https://d1spbj2x7qk4bg.cloudfront.net/en-us/products/totalseq-b-human-tbnk-cocktail-19043?pdf=true&displayInline=true&leftRightMargin=15&topBottomMargin=15&filename=TotalSeq%E2%84%A2-B%20Human%20TBNK%20Cocktail.pdf&v=20220922120634) you should theoretically be able to see some T cells, B cells, NK cells and Monocytes.

jashapiro commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
````diff
@@ -0,0 +1,272 @@
+
+UMAP:
+```{r I map you map we all map for MapMap}
+# We can arbitrarily cluster first just to have some colors in there for funsies
+k <- 10
+
+# perform the clustering
+kclusters <- clusterRows(
+  reducedDim(norm_sce, "PCA"),
+  KmeansParam(centers = k)
+)
````
I'm not sure we need to do clustering at this stage, but if we do, we should do a graph method, as that is much more standard & probably better for these data. On second thought, probably it is worth doing, then we can wave at the cluster/cell type correspondences (or lack therof).

jashapiro reviewed (commented):
Thanks for getting started on this! This looks like a really good start, and just the kind of thing I was hoping to see for a first pass.

As I am at the airport and my brain is a bit fried, I didn't do a deep review, but I did make some comments about things that I saw in a first pass, some of which may make a difference in some of the places where you ran into unexpected results. Or maybe not. 

I do think this dataset is likely to be a good one for the kinds of questions we are after. It seems pretty clean, though I would rather not run into unexpected NA values all over the place. But I think that may be something at our end. I will play with it when I get back if things don't change with other updates.

allyhawkins reviewed (commented):
Just a few comments in addition to Josh's comments, but generally I think we don't want to use the control at all...

sjspielman commented:
Of course just to make it as confusing as possible I did some work on this while @allyhawkins and @jashapiro were reviewing, it seems! I'll integrate these reviews with what I've added. See here: 
[intro-setup.nb.html.zip](https://github.com/AlexsLemonade/training-modules/files/9992989/intro-setup.nb.html.zip)

sjspielman commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
```diff
@@ -0,0 +1,253 @@
+# Filter ADT
+# http://bioconductor.org/books/3.16/OSCA.advanced/integrating-with-protein-abundance.html#applying-custom-qc-filters
+
+controls <- c("IgG1")
```
This might help explain why we have so many `NA`s associated with this code chunk - it's a bad control!

sjspielman commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
```diff
@@ -0,0 +1,253 @@
+# Filter RNA
+droplet_df <- DropletUtils::emptyDrops(
+  counts(pbmc_sce_percellqc), 
+  BPPARAM = BiocParallel::MulticoreParam(4)
```
yes! replacing.

jashapiro commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
```diff
@@ -0,0 +1,340 @@
+norm_sce$celltype_predictions <- preds$pruned.labels
+
+# **heavily** biased towards monocytes and T-cells
+celltype_counts <- table(preds$pruned.labels) %>%
+  as.data.frame() %>%
+  dplyr::rename(
+    celltype = Var1, 
+    n = Freq
+  ) %>%
+  dplyr::arrange(-n)
```
Doing this a bit more tidily?
```suggestion
celltype_counts <- preds %>%
  as.data.frame() %>%
  dplyr::count(pruned.labels) %>%
  dplyr::arrange(desc(n))
```

I also wonder if we might want to start with `colData(norm_sce)` for consistency... since we put the results in there already.

jashapiro commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
````diff
@@ -0,0 +1,340 @@
+                       point_alpha = 0.6, point_size = 0.25)
+```
+
+## ADT "celltyping"
+
+We can leverage the ADTs as protein markers to identify likely celltypes, which can be compared back to the `SingleR` annotations as a form of validation.
````
For pedagogy reasons, I think we will do this the other way around. Antibody tags are the "traditional" way of identifying cell types, so do that first, then show that we can also do it with only RNA.

jashapiro commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
````diff
@@ -0,0 +1,340 @@
+```
+
+
+It might be helpful to hone in on the most prevalent celltypes only for ease of visualization.
+
+```{r}
+# The top 5 cell types account for over 98% of the cells!
+celltype_counts %>%
+  dplyr::slice(1:5) %>%
+  dplyr::summarize(sum(n) / sum(celltype_counts$n))
+
+top_celltypes <- as.character(celltype_counts$celltype[1:5]) 
+
+# Recode a new celltype vector with top 5 separated from "other" using some forcats magic
+norm_sce$celltype_predictions_top <- forcats::fct_lump_n(norm_sce$celltype_predictions, 5)
+
+scater::plotReducedDim(norm_sce, "UMAP", colour_by = "celltype_predictions_top", 
+                       point_alpha = 0.6, point_size = 0.25)
+```
````
reserving judgement about whether we will want to keep this.

sjspielman replied:
moved to the bottom with a note it's maybe too much.

jashapiro commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
````diff
@@ -0,0 +1,340 @@
+rownames(altExp(norm_sce))
+
+# Which ADT is the most prevalent for each cell? 
+max_adt_indices <- apply(
+  logcounts(altExp(norm_sce)),
+  2,
+  which.max
+)
+max_adt <- rownames(altExp(norm_sce))[max_adt_indices]
+
+unique(max_adt)
+
+# Add this into the main SCE experiment for UMAPing
+norm_sce$max_adt <- max_adt
+
+
+# Misc thinking...how can we quantify this?
+## ratio of biggest:\sum(rest)?
+## difference between highest and next highest?
+```
````
I was thinking of a much simpler application at this stage, as some of these markers will appear on more than one cell type: Plot just the expression of individual markers, so something like:

```
scater::plotReducedDim(norm_sce, "UMAP", colour_by = "CD8", 
                       point_alpha = 0.6, point_size = 0.25)
```
Just on its own, this effectively tags the CD8 T cells.

If we want to get more sophisticated, we could make some box/violin/whatever plots showing the expression of these markers by the cluster assignments from unsupervised clustering.

jashapiro reviewed (commented):
Okay, I think this dataset looks like it will work, so I don't want to drag this particular review out too long!

I left a number of comments here, but a lot of them could be addressed in subsequent PRs. 

The major thing I think should be in at this stage include the `renv` additions, and maybe some of the general structure changes (split first, ask quality later) that will feed down later.

But much of the pedagogy will get developed in #566 and #567, so we don't need to solve it all now. But let's get the renv in before we proceed, and then move to #565.

Another question is whether we want to add a second data set for exercises, but I think we should do that in a separate PR as well. (We may need to add an issue for that specifically)

sjspielman commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
```diff
@@ -0,0 +1,340 @@
+
+library(SingleCellExperiment)
+library(magrittr)
+library(SingleR) # annotation
```
In the `renv.lock` at the top-level of `training-modules`, right? Are there other docker-y places this should get added?

jashapiro replied:
Yes, that is the `renv.lock` file to update. Everything else pulls from there.

sjspielman commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
````diff
@@ -0,0 +1,340 @@
+### Initial Mitochondria QC
+
+```{r}
+pbmc_sce_percellqc <- scater::addPerCellQC(
````
> My general recommendation is going to be to not change the name of the object when adding QC stats or otherwise changing metadata, to prevent extra large copies

Normally I agree with this, but in the context of workshops specifically (with the big caveat that this workshop is likely to launch in person instead of remote) this is one of the biggest places I've seen them get tripped up. It's really easy mess up or skip one of the variable overwrites, and as a consequence participants have to back up a lot and re-run bunch of code chunks to get back to a correct/clean version of the object. The time to fix kind of problem this knocks them off course for a while since it almost always needs a breakout room. 

So, I might advocate for - using different variable names but talking about why it's a Choice, and then maybe `rm`ing on some large objects once we arrive at the filtered version? What are your thoughts on that teaching approach?

sjspielman commented:
I've incorporated lots of the reviews here (hopefully all of them!), and this is ready for another look. I also renamed the notebook to `intro-citeseq-setup.Rmd` to better reflect its content.

Here's the rendered version:
[intro-citeseq-setup.nb.html.zip](https://github.com/AlexsLemonade/training-modules/files/10005509/intro-citeseq-setup.nb.html.zip)

sjspielman commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
```diff
@@ -0,0 +1,340 @@
+  pbmc_sce_raw, 
+  subsets = list(mito = mito_genes))
+
+# Filter out 0s, and see before/after dims:
+dim(pbmc_sce_percellqc) 
+pbmc_sce_percellqc <- pbmc_sce_percellqc[, pbmc_sce_percellqc$total > 0]
+dim(pbmc_sce_percellqc) 
```
I removed it but added in a but about `sum` vs. `total`.

sjspielman commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
```diff
@@ -0,0 +1,340 @@
+
+# Normalize ADT
+
+# Calculate baseline:
+# Try out control-based normalization here 
+#  http://bioconductor.org/books/3.16/OSCA.advanced/integrating-with-protein-abundance.html#control-based-normalization
+control_sf <- scater::librarySizeFactors(altExp(norm_sce), subset_row=controls)  # `controls` was previously defined during filtering
+sum(control_sf == 0)/length(control_sf) # _HOLY WOW_ - virtually everything is 0.......
```
This code is gone, and the negative control antibody will be on its way out soon!

sjspielman commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
````diff
@@ -0,0 +1,340 @@
+UMAP:
+```{r I map you map we all map for MapMap}
+# Perform graph clustering first
+nn_k <- 15
````
20, louvain, and parallel ✔️  ✔️  ✔️

sjspielman commented on scRNA-seq-advanced/setup/intro-setup.Rmd:
````diff
@@ -0,0 +1,340 @@
+
+
+```{r}
+ref_data <- celldex::HumanPrimaryCellAtlasData(ensembl = TRUE) # pick one for now to start
````
😬

sjspielman commented:
Final comment, having some weirdness with my `renv` locally (involving my lack of GCC, only clang right now) so if someone else wants to add some commits to this branch with the packages `miQC`, `celldex`, and `SingleR` go for it.

jashapiro commented:
One other thing is to fix all current spellcheck errors before merging... You can download the last check's results here: https://github.com/AlexsLemonade/training-modules/actions/runs/3464338702 under "Artifacts" or generally by looking at the "Run Spell check" chunk in the failing workflow, i.e. https://github.com/AlexsLemonade/training-modules/actions/runs/3464338702/jobs/5785744713

I recommend that any _code_ items that end up triggering the spell check not be added to [the dictionary](https://github.com/AlexsLemonade/training-modules/blob/master/components/dictionary.txt), but instead be enclosed by backticks.

jashapiro commented:
> @allyhawkins @jashapiro Figured out the IgG1 - it's a mouse isotype negative control, so zeros are expected. Maybe the people who generated the dataset also had some mice libraries floating around and wanted to control for contamination or something. We should probably do something up front in the notebook here - I'm thinking pull its counts out, check that they are zero with a histogram or quick `summary()`, feel satisfied, and remove the tag from the SCE.
> 
> Edit: We can/should(?) also remove all the cells with non-0 `IgG1` counts before purging the tag altogether.

I am not sure we will want to completely remove cells with the tag, as false mapping/background effects are still possible. Any with high counts, yes, but I haven't looked at the distribution. This we can address as we write the instruction notebook.

Can you included a reference for where you found out it was a mouse control?

Also, generally, adding link to the dataset info page would be great... I found [this one](https://support.10xgenomics.com/single-cell-gene-expression/datasets/6.0.0/10k_PBMCs_TotalSeq_B_3p?), but if you had another you were drawing from, add that

allyhawkins commented on scRNA-seq-advanced/setup/intro-citeseq-setup.Rmd:
```diff
@@ -0,0 +1,410 @@
+  dplyr::pull(gene)
+
+# let's have a quick look at the raw SCE and genes:
+pbmc_sce_raw
+mito_genes
```
Minor, but for teaching I would separate these chunks and put the `pbmc_sce_raw` call to print out the SCE directly after reading in the data. And then in a new chunk do the mito file.

allyhawkins replied:
And on that note, we might want to reorganize in general, so that the creation of the `mito_genes` variable actually happens before `addPerCellQC` instead of all the way up here.

jashapiro replied:
Yes, I expect we will split the up much more for teaching!

allyhawkins commented on scRNA-seq-advanced/setup/intro-citeseq-setup.Rmd:
````diff
@@ -0,0 +1,410 @@
+
+```{r}
+# We can see this in the `Type` column in the rowData() where the creators of this dataset
+#  stored experiement information
````
```suggestion
#  stored experiment information
```

allyhawkins commented on scRNA-seq-advanced/setup/intro-citeseq-setup.Rmd:
````diff
@@ -0,0 +1,410 @@
+```
+
+These are all human surface proteins except `IgG1`, which is a mouse isotype that was used as a control. 
+We'd, if there's no contamination, that there are 0 counts for this protein since the data is human:
````
```suggestion
If there's no contamination, there should be 0 counts for this protein since the data is human:
```

allyhawkins commented on scRNA-seq-advanced/setup/intro-citeseq-setup.Rmd:
````diff
@@ -0,0 +1,410 @@
+
+```{r}
+miqc_model <- miQC::mixtureModel(filtered_sce) 
+filtered_sce <- miQC::filterCells(filtered_sce,
````
I think we might to explore miQC a bit and look at some of the outputs. I think this is another PR, but just wanted to comment that I don't want to leave miQC at this simplicity and I think we should explore it a bit more.

jashapiro replied:
Agreed!

allyhawkins commented on scRNA-seq-advanced/setup/intro-citeseq-setup.Rmd:
```diff
@@ -0,0 +1,410 @@
+But, be aware that RNA and ADTs are treated differently. 
+As always we recommend [Orchestrating](ttp://bioconductor.org/books/3.16/OSCA.advanced/integrating-with-protein-abundance.html).
+
+We will focus on only removing 0 count cells.
```
Is `cleanTagCounts` only removing 0 count cells? I think it's a bit more complex than that since they compute an ambient factor for each cell and remove based on that. I know we haven't added the full teaching explanation yet, but I want to make sure that we don't include false information. Maybe just for now we should mention that it uses an ambient factor to filter cells?

allyhawkins commented on scRNA-seq-advanced/setup/intro-citeseq-setup.Rmd:
````diff
@@ -0,0 +1,410 @@
+
+```{r}
+ref_data <- celldex::HumanPrimaryCellAtlasData(ensembl = TRUE) # pick one for now to start
+preds <- SingleR::SingleR(
````
```suggestion
singleR_predictions <- SingleR::SingleR(
```
super minor, but can we name this something slightly more informative? I'm open to other names too like `celltype_predictions`

allyhawkins commented on scRNA-seq-advanced/setup/intro-citeseq-setup.Rmd:
````diff
@@ -0,0 +1,410 @@
+```
+
+
+It might be helpful to hone in on the most prevalent cell types only for ease of visualization, but this is a bit more advanced to dive into.
````
I actually think it might be worth showing both plots with all the types and then the plot with just the top cell types. Also offers an opportunity for talking about making good visualizations.

allyhawkins reviewed (commented):
Left a few more comments, some of which can definitely be pushed to future PRs. As far as the control antibody... I'm not entirely sure how to deal with it. I think there is some level of low level detection that might be expected so removing all the cells with > 0 counts there might be too stringent. I think I would agree with removing anything with really high levels. After you remove those, I think you treat that antibody like every other antibody and don't treat it like a control, since it's not really a typical control that the functions we are using are expecting. The other option would be to remove the tag, but not the cells that the tag is expressed in. 

Also, in regards to  cell typing, are we planning on also talking about cell typing with RNA data without CITE-seq and without a reference, e.g. using marker genes or gene sets. For this example I think it would be more straight forward, but I wasn't sure if we were planning on also looking at marker genes in the workshop since not everyone will have CITE-seq or have a solid reference they can use for automation.

jashapiro reviewed (dismissed):
With this now passing CI, I am approving this PR as a "setup draft". 

The remaining updates can and should happen in #565, #566, and #567, so whoever is developing those modules should refer to comments in this PR as development proceeds. I will add notes to those PRs to link to this PR for easier finding, though I expect there will be lots more comments there as well as we get further along.

jashapiro commented:
> Can you included a reference for where you found out it was a mouse control?
> 
> Also, generally, adding link to the dataset info page would be great... I found [this one](https://support.10xgenomics.com/single-cell-gene-expression/datasets/6.0.0/10k_PBMCs_TotalSeq_B_3p?), but if you had another you were drawing from, add that.

I dismissed my past review because I really did want to have the links here confirmed. Also worth adding the main page for MALT dataset, as that may be something we want for the exercise notebook!

jashapiro commented:
> > Also, in regards to cell typing, are we planning on also talking about cell typing with RNA data without CITE-seq and without a reference, e.g. using marker genes or gene sets. For this example I think it would be more straight forward, but I wasn't sure if we were planning on also looking at marker genes in the workshop since not everyone will have CITE-seq or have a solid reference they can use for automation.
> 
> Maybe we should open a separate issue to track including this? I think it's a next-step beyond this notebook.

Yes, this would be part of #567

sjspielman commented:
Links are added, some text is cleaned up, and I included some `miQC` default plots to get that exploration going.

allyhawkins reviewed (approved):
LGTM!

jashapiro reviewed (approved):
LGTM!
